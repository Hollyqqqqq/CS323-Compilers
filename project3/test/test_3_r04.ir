FUNCTION sqr :
PARAM v1
t1 := v1 * v1
RETURN t1

FUNCTION cbc :
PARAM v2
t5 := v2 * v2
t4 := t5 * v2
RETURN t4

FUNCTION main :
READ v3
WRITE v3
ARG v3
v4 := CALL sqr
WRITE v4
ARG v3
v4 := CALL cbc
WRITE v4
RETURN #0

