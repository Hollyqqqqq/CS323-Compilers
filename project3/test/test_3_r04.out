Program (1)
  IncludeList (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: sqr
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: i1
        RP
      CompSt (1)
        LC
        DefList (1)
        StmtList (2)
          Stmt (2)
            RETURN
            Exp (2)
              Exp (2)
                ID: i1
              MUL
              Exp (2)
                ID: i1
            SEMI
          StmtList (1)
        RC
    ExtDefList (4)
      ExtDef (4)
        Specifier (4)
          TYPE: int
        FunDec (4)
          ID: cbc
          LP
          VarList (4)
            ParamDec (4)
              Specifier (4)
                TYPE: int
              VarDec (4)
                ID: i2
          RP
        CompSt (4)
          LC
          DefList (1)
          StmtList (5)
            Stmt (5)
              RETURN
              Exp (5)
                Exp (5)
                  Exp (5)
                    ID: i2
                  MUL
                  Exp (5)
                    ID: i2
                MUL
                Exp (5)
                  ID: i2
              SEMI
            StmtList (1)
          RC
      ExtDefList (8)
        ExtDef (8)
          Specifier (8)
            TYPE: int
          FunDec (8)
            ID: main
            LP
            RP
          CompSt (8)
            LC
            DefList (9)
              Def (9)
                Specifier (9)
                  TYPE: int
                DecList (9)
                  Dec (9)
                    VarDec (9)
                      ID: a
                  COMMA
                  DecList (9)
                    Dec (9)
                      VarDec (9)
                        ID: r
                SEMI
              DefList (1)
            StmtList (10)
              Stmt (10)
                Exp (10)
                  Exp (10)
                    ID: a
                  ASSIGN
                  Exp (10)
                    READ
                    LP
                    RP
                SEMI
              StmtList (11)
                Stmt (11)
                  Exp (11)
                    WRITE
                    LP
                    Exp (11)
                      ID: a
                    RP
                  SEMI
                StmtList (12)
                  Stmt (12)
                    Exp (12)
                      Exp (12)
                        ID: r
                      ASSIGN
                      Exp (12)
                        ID: sqr
                        LP
                        Args (12)
                          Exp (12)
                            ID: a
                        RP
                    SEMI
                  StmtList (13)
                    Stmt (13)
                      Exp (13)
                        WRITE
                        LP
                        Exp (13)
                          ID: r
                        RP
                      SEMI
                    StmtList (14)
                      Stmt (14)
                        Exp (14)
                          Exp (14)
                            ID: r
                          ASSIGN
                          Exp (14)
                            ID: cbc
                            LP
                            Args (14)
                              Exp (14)
                                ID: a
                            RP
                        SEMI
                      StmtList (15)
                        Stmt (15)
                          Exp (15)
                            WRITE
                            LP
                            Exp (15)
                              ID: r
                            RP
                          SEMI
                        StmtList (16)
                          Stmt (16)
                            RETURN
                            Exp (16)
                              INT: 0
                            SEMI
                          StmtList (1)
            RC
        ExtDefList (1)
